### PR TITLE
feat: Add support for VPP 21.06

### PR DESCRIPTION
**Pull request by juliusmilan**
This PR adds support for VPP 21.06

The only significant changes were done for natplugin and ipsecplugin.

In natplugin, there were internal changes caused by separation of VPP nat plugin to 2 separate plugins, for ED and EI nat. However for vpp-agent user perspective, the logic and functionality remained the same, just the code was accommodated, to cover two VPP plugins under one roof. In future, when VPP (EI and ED) nat plugins became differ more and more, there might be a need to separate even vpp-agent natplugin to map 1:1 to VPP nat plugins.

In ipsecplugin, IPSecLink message logic changed, now this message serves just for creation of ipsec interface and bunch of fields, for assigning cryptographic information was deprecated for VPP 21.06. Assigning of cryptographic information to the link is now done by separate messages. Unit tests were updated.

### CODE REVIEW

File: cmd/agentctl/commands/report.go [hunk 1]
```diff
@@ -761,9 +761,14 @@ func printErrorStatsTable(out io.Writer, errorStats *govppapi.ErrorStats) {
 	table.SetAutoWrapText(false)
 
 	for _, errorStat := range errorStats.Errors {
+		var valSum uint64 = 0
+		// errorStat.Values are per worker counters
+		for _, val := range errorStat.Values {
+			valSum = valSum + val
```
Comment (VladoLavor) [inline cmd/agentctl/commands/report.go:767]: Optionally you can use `valSum += val`

Comment (juliusmilan) [reply cmd/agentctl/commands/report.go]: done

File: plugins/vpp/natplugin/vppcalls/vpp2106/nat_vppcalls.go [hunk 2]
```diff
@@ -0,0 +1,782 @@
...
+	reply := &vpp_nat_ei.Nat44EiPluginEnableDisableReply{}
+
+	if err := h.callsChannel.SendRequest(req).ReceiveReply(reply); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Enable NAT44 plugin and apply the given set of options.
+func (h *NatVppHandler) EnableNAT44Plugin(opts vppcalls.Nat44InitOpts) error {
```
Comment (VladoLavor) [inline plugins/vpp/natplugin/vppcalls/vpp2106/nat_vppcalls.go:118]: The comment should have the following format `EnableNAT44Plugin ...`. Also, please mention that this VPP version supports NAT_ED & NAT_EI simultaneously, but vpp-agent does not support it yet. (`// TODO ...`)

File: plugins/vpp/natplugin/vppcalls/vpp2106/nat_vppcalls.go [hunk 3]
```diff
@@ -0,0 +1,782 @@
...
+	if opts.EndpointDependent {
+		h.ed = true
+		return h.enableNAT44EdPlugin(opts)
+	} else {
+		h.ed = false
+		return h.enableNAT44EdPlugin(opts)
```
Comment (VladoLavor) [inline plugins/vpp/natplugin/vppcalls/vpp2106/nat_vppcalls.go:124]: You should call `h.enableNAT44EiPlugin` here

**Comment by VladoLavor** [comment]
Run `go mod tidy -v` and commit the diff in order to pass go.mod check

File: plugins/vpp/natplugin/vppcalls/vpp2106/nat_vppcalls.go (hunk 3 above)
Comment (juliusmilan) [reply plugins/vpp/natplugin/vppcalls/vpp2106/nat_vppcalls.go]: done, thanks

File: plugins/vpp/natplugin/vppcalls/vpp2106/nat_vppcalls.go (hunk 2 above)
Comment (juliusmilan) [reply plugins/vpp/natplugin/vppcalls/vpp2106/nat_vppcalls.go:118]: done

**Review by VladoLavor — APPROVED** [review]

File: proto/ligato/vpp/interfaces/interface.proto [hunk 4]
```diff
@@ -257,30 +257,41 @@ message TapLink {
...
+    enum Mode {
+        // point-to-point tunnel
+        POINT_TO_POINT = 0;
+        // point-to multipoint tunnel (supported starting from VPP 20.05)
+        POINT_TO_MULTIPOINT = 1;
+    };
+    // Mode of the IPIP tunnel
+    Mode tunnel_mode = 1;
+
+    // In VPP 21.06 and newer, IPSecLink serves just for creation of the link and thus all of the following
```
Comment (ondrej-fabry) [inline proto/ligato/vpp/interfaces/interface.proto:269]: Please move this comment to the `IPSecLink` message comment.

Comment (juliusmilan) [reply proto/ligato/vpp/interfaces/interface.proto]: done

**Review by ondrej-fabry — APPROVED** [review]

**Pull request merged**